src/plenigo/services/UserManagementService.php

```diff
@@ -0,0 +1,161 @@
+     * @throws PlenigoException In case of communication errors or invalid parameters
+     */
+    public static function registerUser($email, $language = "en") {
+
+        if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+            $clazz = get_class();
+            PlenigoManager::error($clazz, self::ERR_MSG_EMAIL);
+            return null;
+        }
+
+        $map = array(
+            'email' => $email,
+            'language' => $language
+        );
+
+        $url = ApiURLs::USER_MGMT_REGISTER;
+
+        $request = static::postJSONRequest($url, false, $map);
+
+        $curlRequest = new static($request);
+
+        $data = parent::executeRequest($curlRequest, ApiURLs::USER_MGMT_REGISTER, self::ERR_MSG_REGISTER);
+
+        if (isset($data->customerId)) {
+            $result = $data->customerId;
+        } else {
+            $result = "" . $data;
+        }
+
+        return $result;
+    }
+
+    /**
+     * Change email address of an existing user. This functionality is only available for companies with closed user groups.
+     * 
+     * @param string $customerId Customer id of the user to change email address for
+     * @param string $email New email address of user
+     * @return bool TRUE Email address changed
+     * @throws PlenigoException In case of communication errors or invalid parameters
+     */
+    public static function changeEmail($customerId, $email) {
+
+        if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+            $clazz = get_class();
+            PlenigoManager::error($clazz, self::ERR_MSG_EMAIL);
+            return false;
+        }
+
+        $map = array(
+            'email' => $email
+        );
+
+        $url = str_ireplace(ApiParams::URL_USER_ID_TAG, $customerId, ApiURLs::USER_MGMT_CHANGEMAIL);
+
+        $request = static::putJSONRequest($url, false, $map);
+
+        $curlRequest = new static($request);
+
+        parent::executeRequest($curlRequest, ApiURLs::USER_MGMT_CHANGEMAIL, self::ERR_MSG_CHANGEMAIL);
+
+        return true;
+    }
+
+    /**
+     * Create a login token for an existing user. This functionality is only available for companies with closed user groups.
+     * 
+     * @param string $customerId Customer id of the user to create login token for
+     * @return string One time token used to create a valid user session
+     * @throws PlenigoException In case of communication errors or invalid parameters
+     */
+    public static function createLoginToken($customerId) {
+
+        $url = str_ireplace(ApiParams::URL_USER_ID_TAG, $customerId, ApiURLs::USER_MGMT_CREATELOGIN);
+
+        $request = static::postRequest($url);
+
+        $curlRequest = new static($request);
+
+        $data = parent::executeRequest($curlRequest, ApiURLs::USER_MGMT_CREATELOGIN, self::ERR_MSG_CREATELOGIN);
+
+        if (isset($data->loginToken)) {
+            $result = $data->loginToken;
+        } else {
+            $result = "" . $data;
```

Comment:
The $result object just contains the loginToken property so we are returning that. If there is an error or otherwise different response, we fail gracefully by returnin a string representation of the body of the response